Merge from AS

jrb0001 opened the pull request:
Can be found in `merge-as` branch, needs manual merging because GitHub doesn't allow changing the source of a PR.

reaperrr committed:
Remove lint rule for victim scan radii

They're now calculated automatically to use the most sensible value by default. Only people who know what they're doing or fool around will use custom values, and in that case a lint rule might actually just annoy people.

reaperrr committed:
Make projectiles use most sensible blocker scan radius

By default, but allow custom overrides.

reaperrr committed:
Make warheads use the the most sensible victim scan radius

By default, but allow custom overrides.

obrakmann committed:
Allow cancelling an activity without aborting the entire queue

atlimit8 committed:
Remove DisabledOverlay using WithColoredOverlay & remove IDisable from Husk

atlimit8 committed:
Convert ProximityExternalCondition from disable-able to conditional.

atlimit8 committed:
Make RepairsUnits pausable-conditional

atlimit8 committed:
Make Repair activity halt if RepairsUnit is paused or abort if disabled.

atlimit8 committed:
Added PausableConditionalTrait abstract trait

atlimit8 committed:
Added /poweroutage dev command.

Commit:
Global chat name is reset when connecting #12325
removed TrySetNickname(Game.Settings.Player.Name) in GlobalChat ctor overwriting Chat.Nickname. instead of removing, it could be replaced by TrySetNickname(Game.Settings.Chat.Nickname)

atlimit8 committed:
Convert WithIdleOverlay from actor disabled to pausable-conditional.

atlimit8 committed:
WithRepairOverlay into pausable-conditional trait EMP disabled polish.

atlimit8 committed:
Added ts power outage WithIdleOverlay polish.

atlimit8 committed:
Convert AffectedByPowerOutage from disabler to conditional condition granter.

atlimit8 committed:
Removes actor disabled support from WithIdleAnimation.

atlimit8 committed:
Convert WithRearmAnimation from disable to conditional.

atlimit8 committed:
Convert WithRepairAnimation from disable to conditional.

atlimit8 committed:
Added upgrade rule for PauseOnLowPower removal & AffectedByPowerOutage condition granting.

atlimit8 committed:
Convert support powers from actor disabled to pausable-conditional.

reaperrr committed:
Add explicit Reject field to RejectsOrders

reaperrr committed:
Add support for multiple RejectsOrders traits per actor

reaperrr committed:
Add RevealOnDeath trait

Allows revealing shroud on actor death.

reaperrr committed:
Make D2k units reveal shroud on destruction

For 4 seconds.

Commit:
Host can change Team and Spawn of any player. #12936

reaperrr committed:
Add Firestorm Juggernaut to TS

reaperrr committed:
Add StartBurstReport to WeaponInfo

The FS Juggernaut firing sound already contains 3 shots, so without this, players would hear a total of 9(!) shots per salvo.

reaperrr committed:
Add Name ID field to WithSpriteBody

reaperrr committed:
Make WithMoveAnimation multi-WithSpriteBody-compatible

reaperrr committed:
Fix GrantConditionOnDeploy to support multiple sprite bodies

MustaphaTR committed:
Add Harkonnen 3b

MustaphaTR committed:
Add Ordos 3a

Commit:
Fix Service Depot Rally point path finding (+rename ignoredActor)

Commit:
RepairableNearUnitsWithRepairCursor #12361

Commit:
Fix keys scrolling stuck (+ some function double calls)

abcdefg30 committed:
Add support for using launch-game.cmd with parameters

MustaphaTR committed:
Add Ordos 3b

reaperrr committed:
Make FallToEarth kill the actor instead of disposing it

pchote committed:
Package OSX launchers in a disk image.

pchote committed:
Separate release/playtest/git installs.

pchote committed:
Create separate Windows launchers for each mod.

abcdefg30 committed:
Replace Camera.Truk by using ValidStances on RevealsShroud

abcdefg30 committed:
Fix the Colt sometimes shooting even after Tanya was released

abcdefg30 committed:
Remove the hacky MissionColt actor from allies05a

abcdefg30 committed:
Remove outdated values and replace Bullet by InstantHit in allies05a

MustaphaTR committed:
Add Harkonnen 4

pchote committed:
Create separate Linux launchers for each mod.

pchote committed:
Allow non-openra:// Launch.URI prefixes.

pchote committed:
Allow non-openra:// Launch.URI prefixes.

Commit:
Fix ignored 'Failed to save map' #13036

reaperrr committed:
Prevent attacking while in tunnel/underground

Commit:
LAN games discovery

abcdefg30 committed:
Add a IAutoSelctionSizeInfo trait interface

pchote committed:
Fix tunnel top definitions.

reaperrr committed:
Fix armaments/projectiles to aim at closest Target.Positions

Instead of CenterPosition.
TargetablePositions were already used for targeting/attack decisions, but not armaments/projectiles.

reaperrr committed:
Give D2k structures footprint hit-shapes

reaperrr committed:
Move Tiberian Fiend and visceroid sequences to critters.yaml

reaperrr committed:
Move Tib Fiend and visc rules to critters.yaml

pchote committed:
Improve general installer quality:

- Remove uninstaller and readme start menu links
  (they are explicitly discouraged by best practices)
- Explicitly declare installer execution level
- Define additional control panel metadata

reaperrr committed:
Add support for random sounds to AmbientSound

reaperrr committed:
Add FS Tiberium Floater to TS

For better compatibility with FS maps, and more variety for mappers.

reaperrr committed:
Add Tib Floaters to Sunstroke

Commit:
Enhance - unify global chat (with lobby chat)

Commit:
Helis return to nearest base like Planes

pchote committed:
Fix repair cursor visual feedback target.

reaperrr committed:
Merge pull request #12971 from rob-v/LobbyResetingPlayerName

Global chat name is reset when connecting

reaperrr committed:
Merge pull request #13150 from rob-v/SaveMapError

Show 'Failed to save map' error message in dialog box

atlimit8 committed:
Merge pull request #13174 from reaperrr/ts-floater

Add Tiberium Floater to TS

atlimit8 committed:
Merge pull request #13104 from rob-v/RepairNavalUnitsWithRepairCursor

Naval units (RepairableNear) units repairable using Repair cursor

Commit:
Take screenshot in dialogs #13184

reaperrr committed:
Add custom DeathTypes/DamageTypes support to RevealOnDeath

pchote committed:
Keep ObjectCreator ResolveAssembly override bound until disposal.

pchote committed:
Remove unused ObjectCreator constructor.

Enverex committed:
Correct make install command

No-longer "install-all".

reaperrr committed:
Fix D2k repair pad footprint

To match the original.

pchote committed:
Create separate OSX launchers for each mod.

atlimit8 committed:
ActorExts.AppearsFriendlyTo simplification

reaperrr committed:
Merge pull request #13195 from atlimit8/ActorExts-AppearsFriendlyTo-simplification

ActorExts.AppearsFriendlyTo Simplification

atlimit8 committed:
Merge pull request #13181 from pchote/repair-cursor-flash

Fix repair cursor visual feedback target.

jrb0001 committed:
fix build of AS

reaperrr committed:
Merge pull request #13190 from pchote/object-weirdness

Fix ObjectCreator assembly resolving on Windows.

pchote committed:
Merge pull request #13191 from Enverex/patch-1

Correct make install command for Linux

pchote committed:
Merge pull request #13133 from pchote/windows-packaging

Update Windows packaging for per-mod launchers.

daVoodooShuffle committed:
Closes #13153

Modified all expressions to contain a space on either side

Only effects the YAML files that contain config variables:

X:
Y:
Width:
Height:

Commit:
Fix Selecting a map resets player color

reaperrr committed:
Fix TS infantry move speed

E3 speed in original was 4.
JumpJet ground speed in original was 5.

reaperrr committed:
Fix TS infantry sight range

In original TS, E1 had 5, E2 and E3 had 7.

reaperrr committed:
Fix TS engineer health to match original

reaperrr committed:
Fix TS MCV stats to match original

reaperrr committed:
Fix Mobile Sensor Array sight ranges to match original

reaperrr committed:
Remove bogus Turreted from TS amphib APC

IceReaper committed:
Added ReloadSound and ReloadSoundDelay to weapon declarations.

Baxxster committed:
Add support for blacklist response

Adds support for the new blacklist response in servernames

pchote committed:
Reenable and polish broken bridge in Sunstroke.

atlimit8 committed:
Merge pull request #12924 from reaperrr/fix-tunnel-atk

Prevent units in tunnels/underground from attacking

pchote committed:
Match TTNK pre/post deploy sizes and offsets.

pchote committed:
Match LPST pre/post deploy sizes and offsets.

pchote committed:
Fix WRot FieldLoader parsing.

pchote committed:
Allow fixed barrel pitches to be defined in yaml.

pchote committed:
Set the default voxel Scale to match the original game.

atlimit8 committed:
Merge pull request #13202 from reaperrr/UpgradeRejectsOrders

Add Reject field to RejectsOrders for blacklisting orders.

phrohdoh committed:
Merge pull request #13175 from rob-v/GlobalChat

Enhance - unify global chat (with lobby chat)

pchote committed:
Implement "Enter Tunnel" cursor.

reaperrr committed:
Merge pull request #13186 from rob-v/TakeScreenshotInDialogs

Take screenshot in dialogs #13184

reaperrr committed:
Merge pull request #13210 from pchote/polish-deployers

Polish TS deployer animations

GraionDilach committed:
Expose the drop-disguise-when-attacking aspect to yaml.

Also adds Damaged for more usecases.

GraionDilach committed:
Merge remote-tracking branch 'OpenRA/bleed'

pchote committed:
Merge pull request #13138 from pchote/deb-packaging

Update Linux packaging for per-mod launchers.

pchote committed:
Merge pull request #13129 from pchote/osx-packaging

Update OSX packaging for per-mod launchers.

pchote committed:
Overhaul the packaging scripts:

- Ingame mod versions now always match the package version
- Adds macOS compatibility
- Removes trait and lua api docs pending future overhaul
- Individual platform packages can be compiled directly
- Improved error checking
- Removes unnecessary redundancy and indirection

pchote committed:
Add new high resolution D2K mod logo.

GraionDilach committed:
Fix AIDeployHelper blinkdeploy bug.

reaperrr committed:
Make D2k structures reveal shroud on death

obrakmann committed:
Make Transform use child activities

reaperrr committed:
Merge pull request #12955 from atlimit8/RemoveIDisable-part1

Remove IDisable - part 1

pchote committed:
Merge pull request #13204 from IceReaper/reloading-sound

Added ReloadSound and ReloadSoundDelay to weapon declarations.

GraionDilach committed:
Hide cost properties on buildable free actors.

reaperrr committed:
Merge pull request #13211 from pchote/d2k-hires-logo

Add new d2k mod logo

pchote committed:
Merge pull request #12999 from reaperrr/RevealOnDeath

Add RevealOnDeath trait

pchote committed:
Fix jumpjet flying death animations.

reaperrr committed:
Upgrade rule for DeployAnimation removal

reaperrr committed:
Fix TS Nod artillery turret / barrel / weapon offsets

GraionDilach committed:
Implement the possibility of costed support powers.

GraionDilach committed:
Fix RangeMultiplier regression.

pchote committed:
Move GetLoaders to ObjectCreator.

pchote committed:
Resolve assembly paths before loading the FileSystem.

reaperrr committed:
Add debug overlay for custom TargetPositions

pchote committed:
Remove unused leftover oramod loading code.

pchote committed:
Move D2kSoundResources to Mods.D2k dll.

atlimit8 committed:
Added rule upgrade notice for support powers' PauseOnCondition.

GraionDilach committed:
Merge remote-tracking branch 'OpenRA/bleed'

GraionDilach committed:
Merge remote-tracking branch 'GraionDilach/fix-rangemodifier-regression'

GraionDilach committed:
Merge remote-tracking branch 'GraionDilach/costed-supports'

pchote committed:
Remove unused MiniYaml DictFromFile.

pchote committed:
Persist unknown settings.yaml entries across load/save.

Commit:
Number AI players

reaperrr committed:
Added helper utility for finding best hit-shape scan radius

reaperrr committed:
Add IBlocksProjectilesInfo interface

reaperrr committed:
Upgrade rule for victim scan radii changes

The game/engine now calculates the best values automatically and the field names have changed, so instead of renaming, we can just remove them under the assumption that only modders who really know what they're doing will ever use the override.

reaperrr committed:
Refactor Shape.Rectangle.RotateToIsometry

And fixes TS wall and gate shapes.

reaperrr committed:
Merge pull request #12919 from obrakmann/transform-refactor

Refactor Transform to make use of child activities

reaperrr committed:
Merge pull request #13002 from rob-v/LobbyOptionsEditableByHost

Host can change Team and Spawn of any player. #12936

atlimit8 committed:
Merge pull request #13219 from GraionDilach/fix-rangemodifier-regression

Fix RangeMultiplier regression.

reaperrr committed:
Merge pull request #13217 from pchote/jumpjet-death

Fix jumpjet flying death animations

reaperrr committed:
Make attacking actors/turrets face the targeted position

reaperrr committed:
Merge pull request #13162 from pchote/tunnel-enter-cursor

Implement "Enter Tunnel" cursor

pchote committed:
Merge pull request #13214 from GraionDilach/expand-disguise

Expose the drop-disguise-when-attacking aspect to yaml.

reaperrr committed:
Merge pull request #13224 from daVoodooShuffle/fix-13153

Cleaned Up Chrome Expression Whitespacing

pchote committed:
Replace legacy Evaluator with IntegerExpressions.

Commit:
(Lobby)MapPreviewLogic refactoring

pchote committed:
Add KeyInput.IsRepeat.

pchote committed:
Prevent repeated keys from toggling observer shroud selector.

Commit:
Replays with MapPreview (like in Lobby)

pchote committed:
Define RectangularIsometric world coordinate scale along the cell axis.

pchote committed:
Adjust TS subcell positions for new coordinate system.

Commit:
Remove ChatSettings.NickName, use PlayerName always as default

pchote committed:
Merge pull request #12996 from atlimit8/RemoveIDisable-part2

Remove IDisable - part 2

pchote committed:
Merge pull request #13093 from rob-v/ServicePadMoveToRallyPoint

Fix Service Depot Rally point path finding

reaperrr committed:
Merge pull request #13248 from pchote/detect-key-repeat

Prevent repeated keys from toggling observer shroud selector.

reaperrr committed:
Merge pull request #13232 from rob-v/NumberAIPlayers

Number AI players

reaperrr committed:
Merge pull request #13178 from rob-v/ReturnHeliToNearestBase

Helis return to nearest base like Planes

pchote committed:
Implement mod-defined package loaders.

pchote committed:
Fix WithHarvesterOffset.LocalOffset name.

pchote committed:
Update TS LocalOffsets for new coordinate system.

pchote committed:
Fix grenade speed.

pchote committed:
Fix missile characteristics.

GraionDilach committed:
Merge remote-tracking branch 'OpenRA/bleed'

reaperrr committed:
Merge pull request #13156 from MustaphaTR/d2k-harkonnen-3b

 D2K - Add Harkonnen Mission 3b

atlimit8 committed:
Merge pull request #13253 from pchote/rescale-isometric-world-coords

Rescaled isometric world coordinates to measure 1024 along the cell axes.

atlimit8 committed:
Merge pull request #13247 from pchote/widget-evaluator-expressions

Replaced legacy Evaluator with IntegerExpressions.

Commit:
Add Unit stance hotkeys, remove Cycle Stance hotkey

reaperrr committed:
Adapt blocking height of TS walls etc.

to WDist rescaling.

Also, make wall hit-shape height match blocking height and tweak it to
approximate original TS behaviour as much as possible (Tick Tanks could
shoot over them, but only from certain angles/distances).

reaperrr committed:
Remove TS infantry hit-shape height

We should either implement this for all non-blocker actors at once, or not at all. By now we have walls/gates as examples for vertical shape height, so for now there's no reason to leave this on infantry.

abcdefg30 committed:
Merge pull request #13215 from pchote/packaging-simplification

Overhaul packaging scripts

abcdefg30 committed:
Throw a lint error or an exception when no player owns the world

GraionDilach committed:
Fix the AI acting the same at game restart.

abcdefg30 committed:
Don't close the console window when OpenRA closes

abcdefg30 committed:
Throw a more detailed error message when creating invalid frozen actors

reaperrr committed:
Disable DPI scaling on WIndows by default

reaperrr committed:
Merge pull request #13278 from reaperrr/disable-win-dpi

Disable DPI scaling on Windows by default

reaperrr committed:
Merge pull request #13223 from pchote/mod-package-loaders

Unhardcode mod package loaders

reaperrr committed:
Merge pull request #13203 from reaperrr/ts-balance-fixes

Fix some wrong TS unit stats

CH4Code committed:
D2k editor: Moved all appropriate tiles into Cliff-Type-Changer and replaced Sand-Smooth with Cliff-Ends category.

pchote committed:
Work around resolution issues in (legacy) Fullscreen on OSX.

GraionDilach committed:
Fix AI prereqs.

reaperrr committed:
Merge pull request #13282 from pchote/osx-fullscreen

Work around resolution issues in (legacy) Fullscreen on OSX.

Commit:
Fix Player slot when changing map

Commit:
LobbyLogic, ReplayBrowserLogic Map property changed to map field

reaperrr committed:
Move down D2k High-tech factory sprite

This better aligns its visuals with the footprint, hit-shape and targetable positions.

abcdefg30 committed:
Remove the effect from MissionColt

Commit:
Fix map parameter warning as error from PR 13233

pchote committed:
Implement a workaround for the mono 5.0 cursor enumerator bug.

GraionDilach committed:
Merge remote-tracking branch 'OpenRA/bleed'

reaperrr committed:
Move Shape from Health to new HitShape trait

Renamed Shape to Type

reaperrr committed:
Upgrade rule for Health.Shape to HitShape.Type

reaperrr committed:
Lint check and yaml enforcement for HitShape

Now that Health no longer provides a HitShape, actors with Health need at
least one HitShape trait.

pchote committed:
Fix cloak sounds playing on build for initially cloaked actors.

pchote committed:
Improve handling of unknown error codes.

reaperrr committed:
Add TargetableOffsets to HitShape

And remove ITargetablePositions from Building.
Also, added UseFootprintOffsets to replicate the old Building behavior for easier setup of TargetablePositions for buildings.

reaperrr committed:
Add OuterRadius debug overlay to Rectangle and Capsule

Circle obviously doesn't need it now (might change with the introduction of offsets, but we can still add it when that happens).

reaperrr committed:
Fix Rectangle OuterRadius

Rectangle shapes always assumed that either top-left or bottom-right would have the largest distance from actor center, which is wrong, because depending on the position of the shape, top-right or bottom-left can be the furthest away corner as well, potentially resulting in a too small OuterRadius.

atlimit8 committed:
Grant condition by unit stance

pchote committed:
Rework read/write zip file loading/saving.

pchote committed:
Remove legacy FileSystem.OpenPackage(string, IReadOnlyPackage).

pchote committed:
Move classic-mod package loaders and support files to Mods.Cnc.

pchote committed:
Remove ambiguity from FileSystem references.

pchote committed:
Move InstallShieldPackage to Mods.Common.

Commit:
Catch and ignore BeaconLib errors

jrb0001 committed:
Add KillCargo and FlashScreen to PortableChrono

GraionDilach commented:
Note: **This will break a LOT of things and severely affects balance** due to the WDist rescale.

pchote committed:
Fix "game" chat tab mouseover not working from globalchat login tab.

pchote committed:
Fix bogus variable names in widget widths.

abcdefg30 committed:
Always make sure to be on the host actor when repairing

pchote committed:
Catch and ignore exceptions while loading map previews.

pchote committed:
Update error messages.

Commit:
Dispose-stop Beacon on closing server browser

abcdefg30 committed:
Replace the removed SupplyTruck trait in disable-player-experience.yaml

pchote committed:
Replace Lazy trait lookups with INotifyCreated.

pchote committed:
Explicitly implement interfaces in AttackBase.

pchote committed:
Remove mod chooser reference from music installation prompt.

evgeniysergeev committed:
make IsCloseEnoughToBase function virtual

atlimit8 committed:
Split ConditionByStance in AutoTarget

abcdefg30 committed:
Fix the close enough check in Repair.cs

reaperrr committed:
Fix inconsistent rules in preparation of HitShape refactor

- TD C17 and A10 are not targetable and therefore technically invulnerable, so we should remove Health and Armor
- D2k ^carryall.colorpicker and TS ^mmch.colorpicker shouldn't have Health
- TS crash sites are supposed to be indestructible map deco as per original (they had "Immune=yes")

IceReaper committed:
Fixed PNGLoader crash with palette containing less than 256 colors.

MustaphaTR committed:
Add 8PLAY1

MustaphaTR committed:
Add 8PLAY2

MustaphaTR committed:
Add 8PLAY4

MustaphaTR committed:
Add 8PLAY6

IceReaper committed:
Allow load screens to override StartGame.

obrakmann committed:
Remove tooltips from map preview in MP server browser

obrakmann committed:
Hide tooltips for unoccupied spawnpoints in the replay browser

hartmark committed:
Change so Attack* calls Created on base class

hartmark committed:
Add myself to authors

hartmark committed:
INotifyCreated.Created now calls base properly

pchote committed:
Add engine VERSION file.

pchote committed:
Use engine version for master server queries.

pchote committed:
Use engine version in crash logs.

pchote committed:
Use ContentInstallerMod in MainMenuLogic.

pchote committed:
Ship VERSION with binary releases.

GraionDilach committed:
Merge remote-tracking branch 'OpenRA/bleed'

GraionDilach committed:
Fix AS compliation issues.

/facepalm.

GraionDilach committed:
Merge commit '46d0157c2ec3f3c4fe0d54e4d47d2cf6fe5d2695'

jrb0001 commented:
Changing the source of a PR seems to be impossible so everything is done on #13.
